packages/react-native/ReactCommon/react/renderer/scheduler/SurfaceHandler.cpp

```diff
@@ -233,75 +233,14 @@ Size SurfaceHandler::measure(
   return rootShadowNode->getLayoutMetrics().frame.size;
 }
 
-std::shared_ptr<const ShadowNode> SurfaceHandler::dirtyMeasurableNodesRecursive(
-    std::shared_ptr<const ShadowNode> node) const {
-  const auto nodeHasChildren = !node->getChildren().empty();
-  const auto isMeasurableYogaNode =
-      node->getTraits().check(ShadowNodeTraits::Trait::MeasurableYogaNode);
-
-  // Node is not measurable and has no children, its layout will not be affected
-  if (!nodeHasChildren && !isMeasurableYogaNode) {
-    return nullptr;
+void SurfaceHandler::dirtyMeasurableNodes(std::shared_ptr<const ShadowNode> node) const {
+  if (auto* layoutableNode = dynamic_cast<const YogaLayoutableShadowNode*>(node.get())) {
+    // Only cast away const where needed
+    const_cast<YogaLayoutableShadowNode*>(layoutableNode)->dirtyLayout();
```

Comment:
I'm not sure if it's safe to dirty a potentially sealed node.

Reply:
Previously, the node was cloned before marking it as dirty - the new node is always unsealed.

You also can't really check if a node is sealed or not, since that information is preserved only in debug builds.